File: lcPython/389.find-the-difference.py
#
# @lc app=leetcode id=389 lang=python3
#
# [389] Find the Difference
#

# @lc code=start
class Solution:
    def findTheDifference(self, s: str, t: str) -> str:
        list=[]
        for str in s:
            list.append(str)
        for str in t:
            if str not in list:
                return str
            else:
                list.remove(str)
    def the2ndSolution(self, s: str, t: str) -> str:
        #hashmap to store the occurance of each alphabet

# @lc code=end

File: lcJava/389.find-the-difference.java
/*
 * @lc app=leetcode id=389 lang=java
 *
 * [389] Find the Difference
 */

// @lc code=start


class Solution {
      public static void main(String[] args) {
        Solution solution = new Solution();
        
        // Test with sample inputs
        String s1 = "abcd";
        String t1 = "abcde";
        System.out.println(solution.findTheDifference(s1, t1)); // Output: e

        String s2 = "ae";
        String t2 = "aea";
        System.out.println(solution.findTheDifference(s2, t2)); // Output: a
    }

    public char findTheDifference(String s, String t) {
                // an array to strore the number of occurance of each alphabet
                int[] cnt=new int[26];
                for(char c:s.toCharArray()){
                    cnt[c-'a']++;
                }
                for(char c:t.toCharArray()){
                    if(cnt[c-'a']>0){
                        cnt[c-'a']--;
                    }else
                    return c;
                }
                return ' ';
    }
}


// @lc code=end

